### PR TITLE
Fix audio waveform seeking behaviors

### LINKED ISSUE #623: Audio single result page keyboard navigation is broken
## Description

It is not possible to tab out of the waveform component on the audio result page, nor does the waveform respond to keyboard events anymore for seeking or play/pause.

## Reproduction

1. Run `main` and go to http://localhost:8443/audio/632d0076-fdbd-4919-8b8f-b498ef246ee8
2. Use `tab` to reach the waveform
3. Try to "tab out" of the waveform by hitting tab again.
4. Try to seek the audio using your arrow keys.
5. Try to toggle the audio playback using the spacebar.

Note that none of 3-5 will be successful.

## Expectation

All of 3-5 should work. You should be able to tab out of the audio waveform, toggle the playback state, and seek, all from the keyboard.

## Resolution

- [ ] 🙋 I would be interested in resolving this bug.

### LINKED ISSUE #628: Waveform keyboard seeking does not work
## Description

Seeking on the audio waveform via the keyboard does not work. Distinct from #623 which is only about the `tab` navigation, not the seeking.

## Reproduction

1. Run `main` and visit http://localhost:8443/audio/632d0076-fdbd-4919-8b8f-b498ef246ee8
2. Focus the waveform using the keyboard and attempt to seek the file.
3. Note that it does not work.

## Expectation

The keyboard controls built into the waveform component should work.

## Resolution

- [x] 🙋 I would be interested in resolving this bug.

### PR DESCRIPTION
**Pull request by sarayourfriend**
## Fixes

Fixes #628 by @sarayourfriend
Fixes #623 by @sarayourfriend

Replaces https://github.com/WordPress/openverse-frontend/pull/629 which had a coding error fixed in this PR anyway.

## Description

Fixes keyboard navigation and both keyboard and mouse based seeking on the waveform.

## Testing Instructions

Checkout the branch and run `pnpm dev`. Visit an audio result like http://localhost:8443/audio/632d0076-fdbd-4919-8b8f-b498ef246ee8 and test the keyboard and mouse waveform seeking on the result waveform. Also test this on the related audio.

Verify that you can:
1. Seek using the mouse
2. Seek multiple tracks and have them start playing at the correct position
3. Seek using the keyboard
4. Tab out of the waveform when using keyboard navigation

## Checklist

- [x] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [x] My pull request targets the *default* branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [N/A] I added or updated tests for the changes I made (if applicable).
- [N/A] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Comment by zackkrida** [comment]
I'm seeing a bug where if I let one track play, and then start another track, it begins at the seek point of the previous track instead of starting at the beginning. 

This is a tough problem!

**Comment by sarayourfriend** [comment]
@zackkrida Can you share steps to reproduce? I was seeing that behavior on `main` but this branch should fix it. Specifically this check here with the long comment is meant to address that problem:

https://github.com/WordPress/openverse-frontend/pull/633/files#diff-49554fdd9df51cf0c09140e074e486259b2fe8162c41407f3c8cabbd5767ba7dR144

**Comment by zackkrida** [comment]
@sarayourfriend If I click the play button of one track, let it play for x seconds, and then click the play button on another track, it will begin at the x seconds point.

**Comment by sarayourfriend** [comment]
That's really weird. I'm not able to reproduce that behavior at all, but maying I'm doing something different than what you're describing. Here's a video of me trying to reproduce it:

https://user-images.githubusercontent.com/24264157/150011576-4faaddd8-3f2f-4215-b04c-7123dfbb34d7.mov

**Comment by zackkrida** [comment]
It's specific to Chrome! I tested in Firefox and Safari and saw the correct behavior.

**Comment by sarayourfriend** [comment]
Oh how strange! I'm not able to reproduce it on Chrome 97.0.4692.71 (on an Intel mac running macOS 11.6.2).

**Comment by sarayourfriend** [comment]
I do see one difference in Safari though and that is every time I play an audio track the current time is reset to 0 (which was the original behavior before this PR, but I have no idea why it's doing that on Safari and not Firefox or Chrome for me).

These are very strange differences!

**Comment by zackkrida** [comment]
Ah this is getting weird now-we'll probably need to have more folks check this out:

CleanShot 2022-01-18 at 15 28 31@2x

CleanShot 2022-01-18 at 15 28 48@2x

**Comment by obulat** [comment]
I get a portion of the bug that Zack describes: It looks like the second track seek point starts at the point where the first one left off for a split second, but then it corrects itself. It starts playing from the beginning.
audio-start

File: src/components/VAudioTrack/VAudioTrack.vue [hunk 1]
```diff
@@ -158,7 +180,13 @@ export default defineComponent({
...
       if (window?.audioEl) {
         if (window.audioEl.src !== props.audio.url) {
-          window.audioEl.src = props.audio.url // resets position to zero
+          window.audioEl.src = props.audio.url
+          // Set the current time of the audio back to the seeked time
+          // of the waveform/timeline. In the future we might change
+          // this to reset the audio back to 0 anytime another audio
+          // is played but for now this is simpler and requires less
+          // cross-instance communication.
+          window.audioEl.currentTime = currentTime.value
```
Comment (zackkrida) [inline src/components/VAudioTrack/VAudioTrack.vue:189]:
```suggestion
          // Set the current time of the audio back to the seeked time
          // of the waveform/timeline. In the future we might change
          // this to reset the audio back to 0 anytime another audio
          // is played but for now this is simpler and requires less
          // cross-instance communication.
          window.audioEl.currentTime = currentTime.value
          window.audioEl.src = props.audio.url
```

@sarayourfriend this one weird trick fixes all of the problems for me! I haven't tested this + _removing_ your other attempts to fix browser bugs but I will do so now.

**Review by obulat — APPROVED** [review]

**Comment by sarayourfriend** [comment]
@obulat, just FYI @zackkrida and I ended up pairing on this for a little while to try to resolve the bugs and decided that it might require some significant refactors.

@zackkrida how do you feel about merging this PR as is and then trying those refactors in a follow up? Trying to be conscious of the fact that this PR does fix some fundamentally broken flows, even if it's not perfect. The refactors we discussed might fix everything, but so far they've been really difficult to work out how to get right and I'm in a worse spot than this PR is at the moment.

**Review by zackkrida — APPROVED** [review]

**Pull request merged**